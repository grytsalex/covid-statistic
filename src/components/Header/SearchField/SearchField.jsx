import React, { useRef, memo } from "react";
import PropTypes from "prop-types";

import {
  SearchButton,
  SearchFieldWrapper,
  SearchInput,
} from "./styledComponent";
import img from "../../../assets/search.svg";

export const SearchField = memo(({ onChange }) => {
  const inputRef = useRef(null);
  const onInputFocus = () => {
    inputRef.current.focus();
  };

  return (
    <SearchFieldWrapper className={"input-wrapper"}>
      <SearchInput
        name="search"
        placeholder="search..."
        type="text"
        ref={inputRef}
        onChange={onChange}
        className={"input"}
      />
      <SearchButton icon={img} onClick={onInputFocus} />
    </SearchFieldWrapper>
  );
});

SearchField.propTypes = {
  onChange: PropTypes.func,
};
