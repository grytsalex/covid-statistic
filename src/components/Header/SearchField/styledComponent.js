import styled from "styled-components";

export const SearchFieldWrapper = styled.div`
  height: 80px;
  margin: 40px 60px 0 0;
  position: relative;
`;

export const SearchInput = styled.input`
  min-width: 600px;
  height: 80px;
  position: relative;
  border: none;
  display: block;
  font-size: 48px;
  line-height: 56px;
  font-weight: 400;
  outline: none;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
  border-radius: 20px;
  padding-left: 32px;

  &::-webkit-input-placeholder {
    color: #b2b2b2;
    text-transform: capitalize;
  }
`;

export const SearchButton = styled.button`
  background-image: ${({ icon }) => `url(${icon})`};
  width: 30px;
  height: 30px;
  position: absolute;
  right: 22px;
  top: 0;
  border: none;
  background-color: white;
  transform: translateY(90%);
  outline: none;
  cursor: pointer;
  background-repeat: no-repeat;
`;
