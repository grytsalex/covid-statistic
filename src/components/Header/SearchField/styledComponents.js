import styled from 'styled-components';

export const SearchFieldWrapper = styled.div`
  width: 600px;
  height: 80px;
  margin: 40px 60px 0 0;
  position: relative;

  //&:after {
  //  content: ""
  //}
`;

export const SearchInput = styled.input`
  //width: 100%;
  //height: 100%;
  height:80px;
  position: relative;
  border: none;
  display: block;
  font-size: 48px;
  line-height: 56px;
  font-weight: 400;
  outline: none;
  padding-left: 32px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
  border-radius: 20px;

  &::-webkit-input-placeholder {
    color: #B2B2B2;
    text-transform: capitalize;
  }
`;

export const SearchButton = styled.button`
  background-image: ${({ icon }) => `url(${icon})`};
  width: 30px;
  height: 30px;
  position: absolute;
  right: 5px;
  top: 0;
  border: none;
  background-color: white;
  transform: translateY(90%);
  outline: none;
  cursor: pointer;
  background-repeat: no-repeat;
`;