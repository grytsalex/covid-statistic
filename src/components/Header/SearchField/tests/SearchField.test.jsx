import React from "react";
import { SearchField } from "../SearchField";
import { shallow } from "enzyme";

describe("SearchField", () => {
  const commonProps = {
    onChange: jest.fn(),
  };

  it("should match snapshot", () => {
    const component = shallow(<SearchField {...commonProps} />);

    expect(component).toMatchSnapshot();
  });
});
