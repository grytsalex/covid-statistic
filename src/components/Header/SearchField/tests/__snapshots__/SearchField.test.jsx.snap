// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should match snapshot 1`] = `
<styled.div>
  <styled.input
    name="search"
    onChange={[MockFunction]}
    placeholder="search..."
    type="text"
  />
  <styled.button
    icon="test-file-stub"
    onClick={[Function]}
  />
</styled.div>
`;
